runtime/compiler/env/j9methodServer.cpp

```diff
@@ -1570,6 +1546,39 @@ TR_ResolvedJ9JITServerMethod::createResolvedMethodFromJ9MethodMirror(TR_Resolved
    packMethodInfo(methodInfo, resolvedMethod, fe);
    }
 
+bool
+TR_ResolvedJ9JITServerMethod::verifyCache(TR_ResolvedJ9JITServerMethodInfo clientMethodInfo, TR_ResolvedJ9JITServerMethodInfo cachedMethodInfo, TR_YesNoMaybe unresolvedInCache, bool *unresolvedInCP)
+   {
+   // 
+   if (std::get<0>(cachedMethodInfo).remoteMirror == NULL)
+      {
+      return true;
+      }
+   // If a method is interpreted, some of the parameters might differ from the cached one
+   // But since we are only caching for the duration of one compilation, we ignore the discrepancy
+   if (std::get<0>(clientMethodInfo).isInterpreted == std::get<0>(cachedMethodInfo).isInterpreted)
+      {
+      // cached method is interpreted nothing to compare
+      if (std::get<0>(cachedMethodInfo).isInterpreted)
+         {
+         return true;
+         }
+      if (unresolvedInCP)
+         {
+         if (unresolvedInCache == TR_maybe)
+            return true;
+         if ((unresolvedInCache == TR_yes) != (*unresolvedInCP))
+            {
+            fprintf(stderr, "unresolvedInCP is not NULL, client and cache do not match\n");
+            return false;
+            }
+         }
+      return (clientMethodInfo == cachedMethodInfo);
+      }
+   // If there is a mismatch, return false as the method was previously cached as not interpreted
```

Comment:
should be: "If there is a mismatch, return false if the method was cached as compiled".